src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/HardwareIntrinsicILProvider.cs

```diff
@@ -37,14 +37,25 @@ public HardwareIntrinsicILProvider(InstructionSetSupport isaSupport, FieldDesc i
         public override MethodIL GetMethodIL(MethodDesc method)
         {
             TypeDesc owningType = method.OwningType;
-            string intrinsicId = InstructionSetSupport.GetHardwareIntrinsicId(_context.Target.Architecture, owningType);
+            string intrinsicId = InstructionSetSupport.GetHardwareIntrinsicId(_context.Target.Architecture, owningType, out bool unsupportedSubset);
```

Comment:
Is this just trying to replace the method body for a couple of the special cases?

Can we not just have this logic match what RyuJIT does so we can handle the total set of special "constant" cases:
* `NI_IsSupported_True`
* `NI_IsSupported_False`
* `NI_IsSupported_Dynamic`
* `NI_IsSupported_Type`
* `NI_Throw_PlatformNotSupportedException`
* `NI_Vector_GetCount`